spec/models/order_article_spec.rb

```diff
@@ -117,6 +117,147 @@ def goa_reload
 
   end
 
+  describe 'articel price' do
+    let(:admin) { create :user, groups:[create(:workgroup, role_finance: true)] }
+    let(:article) { create :article }
+    let(:order1) { create :order, article_ids: [article.id] }
+    let(:order2) { create :order, article_ids: [article.id] }
+    let(:order3) { create :order, article_ids: [article.id] }
+    let(:order4) { create :order, article_ids: [article.id] }
+    let(:oa1) { order1.order_articles.first }
+    let(:oa2) { order2.order_articles.first }
+    let(:oa3) { order3.order_articles.first }
+    let(:oa4) { order4.order_articles.first }
+    let(:go1) { create :group_order, order: order1 }
+    let(:go2) { create :group_order, order: order2 }
+    let(:go3) { create :group_order, order: order3 }
+    let(:go4) { create :group_order, order: order4 }
+    let(:goa1) { create :group_order_article, group_order: go1, order_article: oa1 }
+    let(:goa2) { create :group_order_article, group_order: go2, order_article: oa2 }
+    let(:goa3) { create :group_order_article, group_order: go3, order_article: oa3 }
+    let(:goa4) { create :group_order_article, group_order: go4, order_article: oa4 }
+
+    def price_attributes(price)
+      { price: price, unit_quantity: 1 }
+    end
+
+    it 'is correct for open orders' do
+      expect(order1.order_articles.first.price.price).to eq article.price
+      expect(order2.order_articles.first.price.price).to eq article.price
+      expect(order3.order_articles.first.price.price).to eq article.price
+    end
+
+    it 'changs only for open orders when the article is changed' do
+      old_article_price = article.price
+      new_article_price = old_article_price + 1
+      order1.finish!(admin)
+      article.update_attribute(:price, new_article_price)
+      oa1.reload
+      oa2.reload
+      expect(oa1.price.price).to eq old_article_price
+      expect(oa2.price.price).to eq new_article_price
```

Comment:
I wonder if it would be good to test `article_prices` behaviour too, e.g. `current_article_price` object equality.

Reply:
i don't know what to test there. i don't care what happens exactly with `article_prices`

Reply:
Well, in my mind the `article_prices` make up a history of the article prices (perhaps for future statistics purposes). In that sense it could be good to test that the actual `article_price` object is carried over to the `order_article` when the order closes, and that a new `article_price` object is created when the article's price is updated.
I'm ok with leaving that aside right now, it's already an improvement to have this.